src/ipc/ipc-zephyr.c

```diff
@@ -184,6 +184,11 @@ void ipc_platform_send_msg_direct(const struct ipc_msg *msg)
 	intel_adsp_ipc_send_message_emergency(INTEL_ADSP_IPC_HOST_DEV, hdr->pri, hdr->ext);
 }
 
+int ipc_platform_poll_is_host_ready(void)
+{
+	return intel_adsp_ipc_is_complete(INTEL_ADSP_IPC_HOST_DEV);
+}
```

Comment:
I'm too late to the review of the Zephyr side change, so I'm not going to block this here, but, but .... this does feel like a very common problem and problem that is typically solved without polling. I.e.
  - get PM resource (prevent power gaing)
  - send message
  - wait for ack
  - when ack received, release PM resource

But yeah, this kind of implements the same thing with a SOF side API, so this is probably good addition at this point.